src/sherpa_ai/utils.py

```diff
@@ -300,3 +304,19 @@ def check_url(url):
     else:
         return True
     
+def extract_numbers_from_text(text):
```

Comment:
Add unit tests to assure yourself this regex pattern works against the types of numbers you are expecting to match